src/plonk/permutation/prover.rs

```diff
@@ -50,103 +59,146 @@ impl Argument {
     ) -> Result<Committed<C>, Error> {
         let domain = &pk.vk.domain;
 
-        // Goal is to compute the products of fractions
-        //
-        // (p_j(\omega^i) + \delta^j \omega^i \beta + \gamma) /
-        // (p_j(\omega^i) + \beta s_j(\omega^i) + \gamma)
-        //
-        // where p_j(X) is the jth column in this permutation,
-        // and i is the ith row of the column.
-
-        let mut modified_values = vec![C::Scalar::one(); params.n as usize];
-
-        // Iterate over each column of the permutation
-        for (&column, permuted_column_values) in self.columns.iter().zip(pkey.permutations.iter()) {
-            let values = match column.column_type() {
-                Any::Advice => advice,
-                Any::Fixed => fixed,
-                Any::Instance => instance,
-            };
-            parallelize(&mut modified_values, |modified_values, start| {
-                for ((modified_values, value), permuted_value) in modified_values
-                    .iter_mut()
-                    .zip(values[column.index()][start..].iter())
-                    .zip(permuted_column_values[start..].iter())
-                {
-                    *modified_values *= &(*beta * permuted_value + &*gamma + value);
-                }
-            });
-        }
-
-        // Invert to obtain the denominator for the permutation product polynomial
-        modified_values.batch_invert();
+        // How many columns can be included in a single permutation polynomial?
+        // We need to multiply by z(X) and (1 - (l_last(X) + l_blind(X))). This
+        // will never underflow because of the requirement of at least a degree
+        // 3 circuit for the permutation argument.
+        assert!(pk.vk.cs.degree() >= 3);
+        let chunk_len = pk.vk.cs.degree() - 2;
```

Comment:
`chunk_len = pk.vk.cs.degree()` still works for me. Is this expected?

Reply:
That suggests there aren't any tests that have enough columns to trigger the chunking.

Reply:
Updating the test to force there to be multiple sets.